uefi/Cargo.toml

```diff
@@ -19,7 +19,11 @@ logger = []
 # were observed on the VirtualBox UEFI implementation (see uefi-rs#121).
 # In those cases, this feature can be excluded by removing the default features.
 panic-on-logger-errors = []
+# Enables all generic functionality that requires unstable Rust language features. To use
+# features behind this feature gate, you usually need a nightly toolchain.
 unstable = []
+# Like the `unstable` feature but specific to `alloc`-related functionality.
+unstable_alloc = ["unstable", "alloc"]
```

Comment:
Is a separate feature needed? In https://github.com/rust-osdev/uefi-rs/pull/584#discussion_r1031892493 you suggested having a generic `unstable` feature for all the non-stable stuff, which I think makes sense.

Reply:
Eventually, we might end up with

```rust
#![cfg(any(all(feature = "unstable", feature = "alloc"), all(feature = "unstable", feature = "foo")))]
```

and I'd like to prevent that. Without it, we'd need `#[cfg(all(feature = "alloc", feature = "unsafe"))]` at multiple places..

What do you think? @nicholasbishop

Reply:
We could also structure it a little different so that users will only need to activate `unstable` but we still can use "combined features". 

For example, `unstable = ["unstable_alloc"]` and `unstable_alloc = ["alloc"]`? But in that case, `unstable` would force users to enable `alloc`..

Reply:
I agree the cfg syntax gets quite ugly as it gets longer, but all the same it's an internal detail that doesn't affect end users, and I wouldn't expect long ones to be needed all that often. The first example you gave is indeed quite hard to read, but I don't think `#[cfg(all(feature = "alloc", feature = "unsafe"))]` is all that bad.

If in the future we find it's getting too unwieldy, we could also use macros to take out some of the boiler plate. But for now I'd rather start simple with just the `unstable` feature and only worry about complex cases if and when we encounter them.

Reply:
> I agree the cfg syntax gets quite ugly as it gets longer, but all the same it's an internal detail that doesn't affect end users, and I wouldn't expect long ones to be needed all that often. 

Okay, then let's take this way.